TranslationsBox.js

```diff
@@ -64,43 +63,47 @@ const useVLangState = (currVersionObjects, versions) => {
   };
 }
 
-const VersionsBox = ({
+const TranslationsBox = ({
   versions,
-  versionsApiError,
   currVersionObjects,
-  mode,
-  vFilterIndex,
-  recentVFilters,
   segmentRef,
-  setConnectionsMode,
   openFilter,
   openUri,
-  handleOpenURL,
+  openRef,
 }) => {
-  const { themeStr, interfaceLanguage } = useContext(GlobalStateContext);
-  const {
-    vLangState,
-    setVLangState
-  } = useVLangState(currVersionObjects, versions);
+  const {theme, interfaceLanguage} = useGlobalState();
+  const {vLangState, setVLangState } = useVLangState(
+      currVersionObjects,
+      []
+  );
+  const [apiError, setApiError] = useState(false)
+  const loadData = async () => {
+    setApiError(false);
```

Comment:
why are these two lines necessary? presumably the values will be set properly once the api call finishes

Reply:
when moving from segment to segment, i think we don't want the translation of the previous segment to remain until we have the response.